app/Http/Controllers/Apis/Protected/Summit/OAuth2SummitApiController.php

```diff
@@ -435,6 +438,74 @@ public function getAllSummitByIdOrSlugRegistrationStats($id)
         });
     }
 
+    /**
+     * @param $id
+     * @return mixed
+     */
+    public function getAttendeesCheckinsOverTimeStats($id)
+    {
+        return $this->processRequest(function () use ($id) {
+            $summit = $this->repository->getById(intval($id));
+            if (is_null($summit))
+                $summit = $this->repository->getBySlug(trim($id));
+
+            if (is_null($summit)) return $this->error404();
+
+            $current_member = $this->resource_server_context->getCurrentUser();
+            if (!is_null($current_member) && !$current_member->isAdmin() && !$current_member->hasPermissionFor($summit))
+                return $this->error403(['message' => sprintf("Member %s has not permission for this Summit", $current_member->getId())]);
+
+            $group_by = Request::get('group_by');
+            if(!in_array($group_by, IStatsConstants::AttendeesCheckinsAllowedGroupBy))
+                throw new ValidationException("invalid group by");
+
+            $page = 1;
+            $per_page = PaginationValidationRules::PerPageMin;
+
+            if (Request::has('page')) {
```

Comment:
lets create a new trait
ParseAndGetPaginationParams
like
ParseAndGetFilter
and lets encapsulate all this logic there
replace here https://github.com/OpenStackweb/summit-api/blob/c52137708ca99a0eb48036869bd75fe9e79f7091/app/Http/Controllers/Apis/Protected/Summit/Traits/ParametrizedGetAll.php#L105 too
method suggestion
static public function getPaginationParams():array{
    // default values
            $page = 1;
            $per_page = is_null($defaultPageSize) ? PaginationValidationRules::PerPageMin : call_user_func($defaultPageSize);

            if (Request::has('page')) {
                $page = intval(Request::get('page'));
            }

            if (Request::has('per_page')) {
                $per_page = intval(Request::get('per_page'));
            }

return [$page, $per_page];
}

and the using the trait u could do

use ParseAndGetPaginationParams;

// somewhere bellow

list($page, $per_page) = self::getPaginationParams();

Reply:
done